test_vm/src/lib.rs

```diff
@@ -424,6 +435,38 @@ impl<'bs> VM<'bs> {
     pub fn take_invocations(&self) -> Vec<InvocationTrace> {
         self.invocations.take()
     }
+
+    /// Checks the state invariants and returns broken invariants.
+    pub fn check_state_invariants(&self) -> anyhow::Result<MessageAccumulator> {
+        let actors = Hamt::<&'bs MemoryBlockstore, Actor, BytesKey, Sha256>::load(
```

Comment:
This should be fine since it appears to only be called after full message sends, but you should probably do a checkpoint() for safety to make sure that the actors hamt includes all of the latest cached changes.